common/scala/src/main/scala/org/apache/openwhisk/core/yarn/YARNComponentActor.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.openwhisk.core.yarn
+
+import akka.actor.{Actor, ActorSystem}
+import akka.http.scaladsl.model.{HttpMethods, StatusCodes}
+import akka.stream.ActorMaterializer
+import org.apache.openwhisk.common.Logging
+import org.apache.openwhisk.core.entity.ExecManifest.ImageName
+import org.apache.openwhisk.core.yarn.YARNComponentActor.{CreateContainerAsync, RemoveContainer}
+
+import scala.concurrent.ExecutionContext
+
+//Submits create command to YARN and submits decommission command to YARN
+object YARNComponentActor {
+  case object CreateContainerAsync
+  case class RemoveContainer(component_instance_name: String)
+}
+class YARNComponentActor(actorSystem: ActorSystem,
+                         logging: Logging,
+                         yarnConfig: YARNConfig,
+                         serviceName: String,
+                         imageName: ImageName)
+    extends Actor {
+
+  implicit val as: ActorSystem = actorSystem
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
+  implicit val ec: ExecutionContext = actorSystem.dispatcher
+
+  var containerCount: Int = 0
+
+  def receive: PartialFunction[Any, Unit] = {
+    case CreateContainerAsync =>
+      sender ! createContainerAsync
+
+    case RemoveContainer(component_instance_name) =>
+      sender ! removeContainer(component_instance_name)
+
+    case input =>
+      throw new IllegalArgumentException("Unknown input: " + input)
+      sender ! false
+  }
+
+  def createContainerAsync(): Unit = {
+
+    logging.info(this, s"Using YARN to create a container with image ${imageName.name}...")
+
+    val body = "{\"number_of_containers\":" + (containerCount + 1) + "}"
```

Comment:
this is ok although you could also use `JsObject("number_of_containers" -> containerCount + 1).toJson).compactPrint`

Reply:
Makes sense. I changed it to JsObject.